analyzers/src/SonarAnalyzer.Common/SymbolicExecution/Roslyn/Extensions/IOperationExtensions.cs

```diff
@@ -99,6 +100,23 @@ public static IOperation UnwrapConversion(this IOperation operation)
             return operation;
         }
 
+        /// <summary>
+        /// Returns the argument value corresponding to <paramref name="parameterName"/>. For <see langword="params"/> parameter an
+        /// IArrayCreationOperation is returned.
+        /// </summary>
+        public static IOperation ArgumentValue(this ImmutableArray<IOperation> arguments, string parameterName)
+        {
+            foreach (var argument in arguments)
```

Comment:
I would be okay with the change and agree that this is not the hot path. I implemented it this way because:

* ImmutableArray has an allocation-free Enumerator (it has two implementations: `public struct Enumerator` and `private class EnumeratorObject : IEnumerator<T>, IEnumerator, IDisposable`) where the struct has a comment saying *It is important that this enumerator does NOT implement System.IDisposable. We want the iterator to inline when we do foreach and to not result in a try/finally frame in the client.*
* FirstOrDefault allocates two objects (The `EnumeratorObject` and the capture-class for `parameterName`)
* This is a helper function, and from my point of view, these should be efficient because
  * They will be used in all kinds of places and might become a hot path in the future
  * We don't look at them often, and they should be treated just like another public API we use